src/utf-8.cpp

```diff
@@ -178,8 +178,9 @@ const char8* advance_lsp_characters_in_utf_8(string8_view utf_8,
   return utf_8.data() + (c - utf_8_copy.data());
 }
 
-std::ptrdiff_t count_lsp_characters_in_utf_8(padded_string_view utf_8,
-                                             int offset) noexcept {
+std::ptrdiff_t count_lsp_characters_in_utf_8(
+    padded_string_view utf_8, int offset,
+    bool count_utf_16_code_units_in_offset /*= true*/) noexcept {
```

Comment:
This count_utf_16_code_units_in_offset flag makes no sense in isolation.

This function's name (count_lsp_characters_in_utf_8) specifically counts LSP characters. If count_utf_16_code_units_in_offset is false, then the function counts Unicode code points, not LSP characters.

We should make a completely different function (count_code_points_in_utf_8?) instead of hijacking this function. That function can be a copy-paste (and simplification) of this function. (It can be implemented more efficiently in the future.)

Reply:
c4c96eed6962df95b3bb556e81dba495fdaaf0bc